conf/parse_test.go

```diff
@@ -25,6 +25,17 @@ func test(t *testing.T, data string, ex map[string]any) {
 	if !reflect.DeepEqual(m, ex) {
 		t.Fatalf("Not Equal:\nReceived: '%+v'\nExpected: '%+v'\n", m, ex)
 	}
+
+	// Also test ParseWithChecks
+	n, err := ParseWithChecks(data)
+	if err != nil {
+		t.Fatalf("Received err: %v\n", err)
+	}
+	if n == nil {
+		t.Fatal("Received nil map")
+	}
+
+	// DeepEqual will fail due to pedantic mode creating a different map.
```

Comment:
I think we can use JSON here to confirm that the maps are equivalent, can you cherry pick this commit into your branch?
https://github.com/nats-io/nats-server/commit/066cf38d40d87757a718279ad029397bbae204b1